### PR TITLE
[Pavlo Turchyn] UIT-22_Develop_cookie_utility

### PR DESCRIPTION
**Pull request by FabianMoretti**

### CODE REVIEW

File: src/app/core/services/cookie.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CookieService {
+
+  constructor() { }
+
+  set(name: string, value: string, expires?: number | Date, path?: string, domain?: string, secure?: boolean) {
+    let cookieString = name + '=' + value + ';';
```
Comment (dverbovyi) [inline src/app/core/services/cookie.service.ts]: use string interpolation

File: src/app/core/services/cookie.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    if (expires) {
+      if (typeof expires === 'number') {
+        const dateExpires = new Date(new Date().getTime() + expires * 1000 * 60 * 60 * 24);
+        cookieString += 'expires=' + dateExpires.toUTCString() + ';';
```
Comment (dverbovyi) [inline src/app/core/services/cookie.service.ts]: this string is duplicated in else

**Review by dverbovyi — CHANGES_REQUESTED** [review]
@FabianMoretti nice job

please fix 2 minor comments and let's merge it

File: src/app/core/services/cookie.service.ts (hunk 2 above)
Comment (FabianMoretti) [reply src/app/core/services/cookie.service.ts]: These stings are different. If the type of 'expires' is a number, for example, the number of days during which the cookie should exist, then we convert this data in seconds and add it to the current date, and already after that we convert the result into a string type. If the 'expires' type is a date then we only convert it into a string.

Comment (dverbovyi) [reply src/app/core/services/cookie.service.ts]: @FabianMoretti oh really? what's stopping you to create dynamic expires varieble?

```js

const _expires = typeof expires === 'number' ? new Date(new Date().getTime() + expires * 1000 * 60 * 60 * 24) : expires;

cookieString += `expires=${_expires.toUTCString()};`;
```

File: src/app/core/services/cookie.service.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+export class CookieService {
+
+  constructor() { }
+
+  set(name: string, value: string, expires?: number | Date, path?: string, domain?: string, secure?: boolean) {
+    let cookieString = `${name}=${value};`;
+    if (expires) {
+      if (typeof expires === 'number') {
+        const dateExpires = new Date(new Date().getTime() + expires * 1000 * 60 * 60 * 24);
+        cookieString += `expires=${dateExpires.toUTCString()};`;
```
Comment (dverbovyi) [inline src/app/core/services/cookie.service.ts]: not fixed, see comment above

**Review by dverbovyi — CHANGES_REQUESTED** [review]

**Review by dverbovyi — APPROVED** [review]

**Pull request merged**